pkg/agent/inteface.go

```diff
@@ -76,6 +76,19 @@ type AgentAddonOptions struct {
 	// SupportedConfigGVRs is a list of addon supported configuration GroupVersionResource
 	// each configuration GroupVersionResource should be unique
 	SupportedConfigGVRs []schema.GroupVersionResource
+
+	// AgentDeployClusterInformerFilter defines the filter function for the cluster informer when deploying the addon
```

Comment:
the filter func to trigger the agent deploy/redploy when cluster info is changed.

Reply:
fixed